src/services.py

```diff
@@ -143,3 +154,47 @@ def __init__(self, Security=Security):
     def get_all(self):
         with session_scope() as session:
             return [sec.asdict() for sec in session.query(self.Security).all()]
+
+
+class LinkedinService:
+    def __init__(self, User=User):
+        self.User = User
+
+    @validate_input(LINKEDIN_CODE_RULE)
+    def get_user_data(self, code):
+        token = self.get_token(code)
+        user_email = self.get_user_email(token)
+        user_data = {"user_email": user_email}
+        return user_data
+
+    def get_token(self, code):
+        token = requests.post(
+            "https://www.linkedin.com/oauth/v2/accessToken",
+            headers={"Content-Type": "x-www-form-urlencoded"},
+            params={
+                "grant_type": "authorization_code",
+                "code": code,
+                "redirect_uri": APP_CONFIG.get("REDIRECT_URI"),
```

Comment:
I think better if we pass in these params here instead of getting it from the `APP_CONFIG`. So presumably in the API you want to pass in `request.app['REDIRECT_URI']` etc.

That way, the app config is only tied to the app, and there can be less coupling.